seqr/views/utils/dataset_utils.py

```diff
@@ -265,18 +266,23 @@ def _get_unmatched_error(remaining_sample_ids):
 
 def _match_and_update_rna_samples(
     projects, user, sample_project_tuples, data_source, sample_id_to_individual_id_mapping, raise_unmatched_error_template,
+    sample_id_to_tissue_type,
 ):
     def _get_unmatched_error(sample_keys):
         return raise_unmatched_error_template.format(sample_ids=(', '.join(sorted([sample_id for sample_id, _ in sample_keys]))))
 
-    samples = Sample.objects.select_related('individual__family__project').filter(
+    query = Q(
         individual__family__project__in=projects,
         sample_type=Sample.SAMPLE_TYPE_RNA,
         dataset_type=Sample.DATASET_TYPE_VARIANT_CALLS,
         sample_id__in={sample_id for sample_id, _ in sample_project_tuples},
     )
+    tissues = set(sample_id_to_tissue_type.values())
+    query = query & Q(tissue_type__isnull=True) if None in tissues else query & Q(tissue_type__in=tissues)
```

Comment:
This is neater as ` query &= Q(tissue_type__isnull=True) if None in tissues else Q(tissue_type__in=tissues)`

Also, I think it would be safer if the conditional was `tissues == {None}` to make sure you aren't in a weird edge case where one of the values is None